golang/cosmos/x/vtransfer/keeper/keeper.go

```diff
@@ -149,6 +241,14 @@ func (k Keeper) InterceptOnAcknowledgementPacket(
 	if err != nil {
 		return err
 	}
+
+	if StorePacketData && !bytes.Equal(strippedPacket.GetData(), packet.GetData()) {
+		packetStore, packetKey := k.PacketStoreFromOrigin(ctx, types.PacketSrc, packet)
+		if !packetStore.Has(packetKey) {
+			packetStore.Set(packetKey, packet.GetData())
```

Comment:
Why are we remembering the packet data when receiving an ack? Shouldn't we assert we had saved packet data if we rewrote the packet, and delete instead?